src/_pytest/cacheprovider.py

```diff
@@ -372,16 +376,18 @@ def pytest_addoption(parser):
     )
 
 
-def pytest_cmdline_main(config):
+def pytest_cmdline_main(config: Config) -> Optional[Union[int, ExitCode]]:
     if config.option.cacheshow:
         from _pytest.main import wrap_session
 
         return wrap_session(config, cacheshow)
+    return None
 
 
 @pytest.hookimpl(tryfirst=True)
-def pytest_configure(config):
-    config.cache = Cache.for_config(config)
+def pytest_configure(config: Config) -> None:
+    # Type ignored: pending mechanism to store typed objects scoped to config.
```

Comment:
This repeats in several plugins. I have a simple plan for how to do it, but for now I added type-ignores.